src/memberPublicKey.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { getModelForClass, type ReturnModelType } from '@typegoose/typegoose';
+import type { Connection } from 'mongoose';
+
+import { setUpTestDbConnection } from './testUtils/db.js';
+import { makeMockLogging, type MockLogging, partialPinoLog } from './testUtils/logging.js';
+import {
+  MEMBER_MONGO_ID,
+  MEMBER_PUBLIC_KEY_MONGO_ID as PUBLIC_KEY_ID,
+  TEST_OID,
+} from './testUtils/stubs.js';
+import type { ServiceOptions } from './serviceTypes.js';
+import { requireFailureResult, requireSuccessfulResult } from './testUtils/result.js';
+import { getPromiseRejection } from './testUtils/jest.js';
+import { MemberPublicKeyModelSchema } from './models/MemberPublicKey.model.js';
+import { generatePublicKey } from './testUtils/publicKeyGenerator.js';
+import {
+  createMemberPublicKey,
+  deleteMemberPublicKey,
+  getMemberPublicKey,
+} from './memberPublicKey.js';
+import { MemberPublicKeyProblemType } from './MemberPublicKeyProblemType.js';
+
+const publicKey = await generatePublicKey();
+
+describe('member public key', () => {
+  const getConnection = setUpTestDbConnection();
+
+  let mockLogging: MockLogging;
+  let connection: Connection;
+  let serviceOptions: ServiceOptions;
+  let memberPublicKeyModel: ReturnModelType<typeof MemberPublicKeyModelSchema>;
+  beforeEach(() => {
+    mockLogging = makeMockLogging();
+    connection = getConnection();
+    serviceOptions = {
+      dbConnection: connection,
+      logger: mockLogging.logger,
+    };
+    memberPublicKeyModel = getModelForClass(MemberPublicKeyModelSchema, {
+      existingConnection: connection,
+    });
+  });
+
+  describe('createMemberPublicKey', () => {
+    test('Member public key should be created', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: publicKey.toString('base64'),
+          oid: TEST_OID,
+        },
+        serviceOptions,
+      );
+
+      requireSuccessfulResult(memberPublicKey);
+      const dbResult = await memberPublicKeyModel.findById(memberPublicKey.result.id);
+      expect(dbResult?.memberId).toStrictEqual(MEMBER_MONGO_ID);
+      expect(dbResult?.publicKey.toString('base64')).toStrictEqual(publicKey.toString('base64'));
+      expect(mockLogging.logs).toContainEqual(
+        partialPinoLog('info', 'Member public key created', {
+          id: memberPublicKey.result.id,
+        }),
+      );
+    });
+
+    test('Malformed public key should be refused', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: Buffer.from('invalid public key').toString('base64'),
+          oid: TEST_OID,
+        },
+        serviceOptions,
+      );
+
+      requireFailureResult(memberPublicKey);
+      expect(memberPublicKey.reason).toBe(MemberPublicKeyProblemType.MALFORMED_PUBLIC_KEY);
+    });
+
+    test('Record creation errors should be propagated', async () => {
```

Comment:
We don't need this test here: We only need it in functions that could failed due to duplicated records (e.g., `createOrg()`).

Reply:
In `createOrg()` we need two tests indeed because there are two code branches in the `catch` block: the error may or may not be caused by a duplicated record.

But this test doesn't check any branch, because there's no branch to test. There's no `catch`.

Reply:
in that case, there are a lot of places where we can remove this. Shall we remove it?

Reply:
Yes, please. I must've missed those in the reviews.